src/Solvers/multigrid_solver.jl

```diff
@@ -60,25 +83,25 @@ Arguments
 * `reltol, abstol`: Relative and absolute tolerance for convergence of the algorithm.
                     The iteration stops when `norm(A * x - b) < max(reltol * norm(b), abstol)`.
 
-* `amg_algorithm`: Algebraic Multigrid algorithm defining mapping between different grid spacings
-
-!!! compat "Multigrid solver on GPUs"
-    Currently Multigrid solver is only supported on CPUs.
+* `amg_algorithm`: Algebraic Multigrid algorithm defining mapping between different grid spacings.
+                   Options are `RugeStubenAMG()` (default) or `SmoothedAggregationAMG()`.
+                   Note: This keyword is relevant *only* on the CPU.
 """
 function MultigridSolver(linear_operation!::Function,
                          args...;
                          template_field::AbstractField,
                          maxiter = prod(size(template_field)),
-                         reltol = sqrt(eps(eltype(template_field.grid))),
+                         reltol = 0.1*sqrt(eps(eltype(template_field.grid))),
                          abstol = 0,
                          amg_algorithm = RugeStubenAMG(),
+                         matrix = nothing,
```

Comment:
update the docstring to include the `matrix` kwarg

Reply:
Sounds good. Should we let the `linear_operator!` default to something if `matrix` is not nothing?

Reply:
The way it's written now `linear_operation!` is an obligatory arg.
Hm... so how it's written now one could provide, e.g., a `linear_operation!` that corresponds to the Laplacian and a `matrix` kwarg that corresponds to something else. Then the `linear_operation!` is overwritten, right?

I think it could be confusing. How about something like:

```Julia
function MultigridSolver(linear_operation!::Function,
                         args...;
                         template_field::AbstractField,
                         maxiter = prod(size(template_field)),
                         reltol = sqrt(eps(eltype(template_field.grid))),
                         abstol = 0,
                         amg_algorithm = RugeStubenAMG()
                         )

    arch = architecture(template_field)

    matrix = initialize_matrix(arch, template_field, linear_operation!, args...)

    return MultigridSolver(matrix, args...; template_field, maxiter, reltol, abstol, amg_algorithm)
end

function MultigridSolver(matrix::TypeOfMatrix,
                         args...;
                         template_field::AbstractField,
                         maxiter = prod(size(template_field)),
                         reltol = sqrt(eps(eltype(template_field.grid))),
                         abstol = 0,
                         amg_algorithm = RugeStubenAMG()
                         )

    ...
    
end
```

?